frontend-react/src/components/Admin/AdminReceiverDashboard.tsx

```diff
@@ -529,63 +678,165 @@ function ModalInfoRender(props: { subData: AdmConnStatusDataType[] }) {
-                </Grid>
-            </Grid>
+                    <Grid row className={"modal-info-row"}>
+                        <Grid className={"modal-info-label"}>Result:</Grid>
+                        <Grid
+                            className={`modal-info-value ${
+                                dataItem.connectionCheckSuccessful
+                                    ? "success-all"
+                                    : "failure-all"
+                            }`}
+                        >
+                            {dataItem.connectionCheckSuccessful
+                                ? "success"
+                                : "failed"}
+                        </Grid>
+                    </Grid>
 
-            <Grid row className={"modal-info-row"}>
-                <Grid className={"modal-info-label"}>Result message:</Grid>
-                <Grid className={"modal-info-value"}>
-                    {dataItem.connectionCheckResult}
+                    <Grid row className={"modal-info-row"}>
+                        <Grid className={"modal-info-label"}>Started At:</Grid>
+                        <Grid className={"modal-info-value"}>
+                            {formatDate(dataItem.connectionCheckStartedAt)}
+                            <br />
+                            {dataItem.connectionCheckStartedAt}
+                        </Grid>
+                    </Grid>
+
+                    <Grid row className={"modal-info-row"}>
+                        <Grid className={"modal-info-label"}>
+                            Time to complete:
+                        </Grid>
+                        <Grid className={"modal-info-value"}>
+                            {duration(dataItem)}
+                            <br />
+                        </Grid>
+                    </Grid>
+
+                    <Grid row className={"modal-info-row"}>
+                        <Grid className={"modal-info-label"}>
+                            Result message:
+                        </Grid>
+                        <Grid className={"modal-info-value"}>
+                            {dataItem.connectionCheckResult}
+                        </Grid>
+                    </Grid>
                 </Grid>
-            </Grid>
+            ))}
         </GridContainer>
     );
 }
 
+/**
+ * We want to control date picking better than the default control allows.
+ * Reasons:
+ *  - We limit the max range allowed to limit large data requests.
+ *  - We want start AND end picked before the expensive fetch.
+ *  - Picker fields are LARGE and take up a bunch of space.
+ */
+function DateRangePickingAtomic(props: {
+    defaultStartDate: string;
+    defaultEndDate: string;
+    onChange: (props: { startDate: string; endDate: string }) => void;
+}) {
+    const [startDate, setStartDate] = useState<string>(props.defaultStartDate);
+    const [endDate, setEndDate] = useState<string>(props.defaultEndDate); // may be null because it's optional
```

Comment:
I'm going to fix the end date to today (as recommended above), so this will go away. The dates here are ISO strings. I should probably rename the vars so it's super obvious.